lua/plenary/path.lua

```diff
@@ -428,10 +446,21 @@ function Path:_split()
   return vim.split(self:absolute(), self._sep)
 end
 
-function Path:parents()
+function Path:parent()
   return self:absolute():match(string.format('^(.+)%s[^%s]+', self._sep, self._sep))
 end
 
+function Path:parents()
+    local results = {}
+    local cur = self
+    repeat
+        cur = Path:new(cur:parent())
```

Comment:
Do we really want to create a Path object over and over again? The resulting parent should already be `absolute` so we only have to run the regex.

Maybe we can have 
```lua
local _get_parent(abs_path) = (function()
  local formatted = string.format('^(.+)%s[^%s]+', self._sep, self._sep)
  return function(abs_path)
    return abs_path:match(formatted)
  end
end)

function Path:parent()
  return _get_parent(self:absolute())
end

-- your parents using `_get_parent` after first parent was returned with `self:parent()`.